--- a/fissix/main.py
+++ b/fissix/main.py
@@ -295,16 +295,13 @@
     else:
         requested = avail_fixes.union(explicit)
     fixer_names = requested.difference(unwanted_fixes)
-    input_base_dir = os.path.commonprefix(args)
-    if (
-        input_base_dir
-        and not input_base_dir.endswith(os.sep)
-        and not os.path.isdir(input_base_dir)
-    ):
-        # One or more similar names were passed, their directory is the base.
-        # os.path.commonprefix() is ignorant of path elements, this corrects
-        # for that weird API.
-        input_base_dir = os.path.dirname(input_base_dir)
+    if args:
+        input_base_dir = os.path.commonpath(args)
+        if not os.path.isdir(input_base_dir):
+            # args are filenames, use their common parent directory.
+            input_base_dir = os.path.dirname(input_base_dir)
+    else:
+        input_base_dir = ""
     if options.output_dir:
         input_base_dir = input_base_dir.rstrip(os.sep)
         logger.info(
